File: notification-schema-java/src/main/java/uk/gov/defra/tracesx/notificationschema/representation/ControlAuthority.java
package uk.gov.defra.tracesx.notificationschema.representation;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import uk.gov.defra.tracesx.notificationschema.representation.enumeration.IUUOption;

@Builder
@Data
@JsonInclude(Include.NON_EMPTY)
@NoArgsConstructor(access = AccessLevel.PUBLIC)
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public class ControlAuthority {

  private OfficialVeterinarian officialVeterinarian;
  private String customsReferenceNo;
  private Boolean containerResealed;
  private String newSealNumber;
  private String iuuFishingReference;
  private Boolean iuuCheckRequired;
  private IUUOption iuuOption;
}

File: imports-frontend-entities/src/entities/control_authority.js
const _ = require('lodash')
const handler = require('./base/handler')
const OfficialVeterinarian = require('./official_veterinarian')

module.exports = class ControlAuthority {

  constructor(obj = {}) {

    this.controlAuthority = obj.controlAuthority
    this.officialVeterinarian = _.get(obj, 'officialVeterinarian')
        ? new OfficialVeterinarian(obj.officialVeterinarian) : undefined
    this.customsReferenceNo = obj.customsReferenceNo
    this.containerResealed = obj.containerResealed
    this.newSealNumber = obj.newSealNumber
    this.iuuFishingReference = obj.iuuFishingReference
    this.iuuCheckRequired = obj.iuuCheckRequired
    this.iuuOption = obj.iuuOption

    return Object.seal(new Proxy(this, handler))
  }
}
